service/api/service.go

```diff
@@ -6,35 +6,61 @@ import (
 
 	"github.com/ipfs/go-cid"
 	cbornode "github.com/ipfs/go-ipld-cbor"
+	logging "github.com/ipfs/go-log"
 	"github.com/libp2p/go-libp2p-core/crypto"
 	"github.com/libp2p/go-libp2p-core/peer"
 	ma "github.com/multiformats/go-multiaddr"
 	mh "github.com/multiformats/go-multihash"
 	"github.com/textileio/go-threads/cbor"
 	core "github.com/textileio/go-threads/core/service"
 	"github.com/textileio/go-threads/core/thread"
-	"github.com/textileio/go-threads/crypto/symmetric"
+	sym "github.com/textileio/go-threads/crypto/symmetric"
 	pb "github.com/textileio/go-threads/service/api/pb"
 	"github.com/textileio/go-threads/service/util"
+	tutil "github.com/textileio/go-threads/util"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-// service is a gRPC service for a thread service.
-type service struct {
-	s core.Service
+var (
+	log = logging.Logger("threadserviceapi")
+)
+
+// Service is a gRPC service for a thread service.
+type Service struct {
+	ts core.Service
+}
+
+// Config specifies server settings.
+type Config struct {
+	Debug bool
+}
+
+// NewService starts and returns a new service.
+func NewService(ts core.Service, conf Config) (*Service, error) {
```

Comment:
In the case of the threads api server and this service server, it seems like you removed the "sever" and just added a `NewService` method... I'm guessing that is just because the "server" wasn't really providing any functionality other than creating the service?

Reply:
Yep, exactly. After removing the gRPC server, it felt like an unnecessary layer.